src/vm.rs

```diff
@@ -6,11 +6,20 @@ use std::{pin::Pin, time::Instant};
 
 /// The initialized options for the VM
 #[derive(Debug, Copy, Clone)]
-pub struct VmOptions {}
+pub struct VmOptions {
+    pub fault_tolerant: bool,
+}
+impl Default for VmOptions {
+    fn default() -> Self {
+        Self {
+            fault_tolerant: false,
+        }
+    }
+}
 
 impl From<&crate::Options> for VmOptions {
     fn from(_options: &crate::Options) -> Self {
-        Self {}
+        Self { ..Self::default() }
```

Comment:
`crate::Options` has the field `fault_tolerant`, so use that to populate `VmOptions`'s own field